tests/datalab/test_datalab.py

```diff
@@ -980,53 +980,109 @@ def test_pred_probs_precedence(self, pred_probs, random_embeddings):
-    random_idx = np.random.choice(num_examples, num_errors)
-    y[random_idx] += extra_noise
-    error_idx = np.argsort(abs(y - true_y))[-num_errors:]  # get the noisiest examples idx
-
-    # create test set
-    X_test = np.random.random(size=(num_examples, num_features))
-    label_noise = np.random.normal(scale=noise, size=num_examples)
-    y_test = np.dot(X_test, coefficients) + label_noise
-
-    return {
-        "X": X,
-        "y": y,
-        "true_y": true_y,
-        "X_test": X_test,
-        "y_test": y_test,
-        "error_idx": error_idx,
-    }
+class TestDatalabForRegression:
+    @pytest.fixture
+    def regression_data(
+        self, num_examples=200, num_features=3, noise=0.2, error_frac=0.1, error_noise=5
+    ):
+        np.random.seed(SEED)
+        X = np.random.random(size=(num_examples, num_features))
+        coefficients = np.random.uniform(-1, 1, size=num_features)
+        label_noise = np.random.normal(scale=noise, size=num_examples)
+
+        true_y = np.dot(X, coefficients)
+        y = np.dot(X, coefficients) + label_noise
+
+        # add extra noisy examples
+        num_errors = int(num_examples * error_frac)
+        extra_noise = np.random.normal(scale=error_noise, size=num_errors)
+        random_idx = np.random.choice(num_examples, num_errors)
+        y[random_idx] += extra_noise
+        error_idx = np.argsort(abs(y - true_y))[-num_errors:]  # get the noisiest examples idx
+
+        # create test set
+        X_test = np.random.random(size=(num_examples, num_features))
+        label_noise = np.random.normal(scale=noise, size=num_examples)
+        y_test = np.dot(X_test, coefficients) + label_noise
+
+        return {
+            "X": X,
+            "y": y,
+            "true_y": true_y,
+            "X_test": X_test,
+            "y_test": y_test,
+            "error_idx": error_idx,
+        }
+
+    @pytest.fixture
+    def lab(self, regression_data):
+        X, y = regression_data["X"], regression_data["y"]
+        test_df = pd.DataFrame(X, columns=["c1", "c2", "c3"])
+        test_df["y"] = y
+        lab = Datalab(data=test_df, label_name="y", task="regression")
+        return lab
+
+    def test_available_issue_types(self, lab):
+        assert set(lab.list_default_issue_types()) == set(["label"])
+        assert set(lab.list_possible_issue_types()) == set(["label"])
+
+    def test_regression_with_features(self, lab, regression_data):
+        """Test that the regression issue checks finds 40 label issues, based on the
+        numerical features."""
+        X = regression_data["X"]
+        lab.find_issues(features=X)
+        lab.report()
+        summary = lab.get_issue_summary()
+
+        assert "label" in summary["issue_type"].values
+        assert (summary[summary["issue_type"] == "label"]["num_issues"] == 40).all()
+        assert np.isclose(
+            summary[summary["issue_type"] == "label"]["score"].values[0], 0.672974, atol=1e-5
+        )
```

Comment:
can we assert which of the datapoints have issues, best test is assert that checks for high jaccard similarity between set of flagged issues and set of true issues

Reply:
I've updated the dataset for testing and asserted which datapoints have issues. So this should be resolved.

The tricky thing is that the "real issues" might have smaller residuals and are not picked up (leading to a lower Jaccard similarity).

Reply:
Sorry maybe I'm dense and don't see it. Where is the test that is checking for high jaccard similarity between set of flagged issues and set of true issues?

All I see is:

```
assert (summary[summary["issue_type"] == "label"]["num_issues"] == 40).all()
```

which is only testing how many issues were estimated (which we don't care about testing at all actually, we only care about testing which datapoints were flagged as issues and that the overlap between this set and the set of true erroneous values is sufficiently high).

Reply:
The tests checking the high Jaccard similarity were placed behind those lines.

But now the summary assertions have been removed.
I've added some Jaccard Similarity assertions, as well as low FPR assertions as well.